src/core/ext/filters/client_channel/http_proxy.cc

```diff
@@ -42,16 +42,19 @@
  * credentials if present in the 'http_proxy' env var, otherwise leaves it
  * unchanged. It is caller's responsibility to gpr_free user_cred.
  */
-static char* get_http_proxy_server(char** user_cred) {
+static char* get_http_proxy_server(const grpc_channel_args* args, char** user_cred) {
   GPR_ASSERT(user_cred != nullptr);
   char* proxy_name = nullptr;
   char** authority_strs = nullptr;
   size_t authority_nstrs;
-  /* Prefer using 'grpc_proxy'. Fallback on 'http_proxy' if it is not set.
+  /* Check channel arguments in case a per channel proxy is set. Otherwise,
+   * prefer using 'grpc_proxy'. Fallback on 'http_proxy' if it is not set.
    * Also prefer using 'https_proxy' with fallback on 'http_proxy'. The
    * fallback behavior can be removed if there's a demand for it.
    */
-  char* uri_str = gpr_getenv("grpc_proxy");
+  const grpc_arg* arg = grpc_channel_args_find(args, GRPC_ARG_HTTP_PROXY);
```

Comment:
Please use `grpc_channel_args_find_string()` here:

https://github.com/grpc/grpc/blob/e3fb73fec501f2b7b27999ea9b10e41d7ac4ea22/src/core/lib/channel/channel_args.h#L93

Reply:
Changed code to use grpc_channel_args_find_string function.